### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by duskotmane**
# Features 🌟 

- [ ] Create a class `Classroom`.
- [ ] Create the has-many/belongs-to relationship between `Classroom` and `Student`.
- [ ] Create a class `Book`.
- [ ] Create a class `Rental`.
- [ ] Create the many-to-many relationship between `Person` and `Book` using the intermediate class `Rental`.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 class Student < Person
-  attr_reader :classroom
+  attr_accessor :classroom
 
-  def initialize(classroom, id, age, name = 'Unknown', parent_permission: true)
-    super(id, age, name, parent_permission: parent_permission)
+  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
   end
```
Comment (codepantha) [inline student.rb]: - [ ] Please create the `belongs-to` side of the relationship between `Student` and `Classroom`. Please see this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) if you need a refresher

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(rental)
+    @rentals.push(rental)
+  end
```
Comment (codepantha) [inline book.rb]: - [ ] A book has many rentals, while a rental belongs to a book and also belongs to a person. Please initialize the Rental here (`Rental.new`) while passing in the book itself as a parameter to create the new rental achieving the 'has-many' association. 
```ruby
def add_rental(person, date)
  Rental.new(date, self, person)
end
```
#### 🔔 Kindly do the same for the `Person` class
Afterward, the constructor of `Rental` should be modified so `Book` and `Person` are set in it to break the `many-to-many` relationship.

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (codepantha) [inline rental.rb]: - [ ] Please modify the constructor of `Rental` so `Book` and `Person` are set in it to break the many-to-many relationship. Kindly see this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) if you need a refresher 🙏 
```ruby
def initialize(date, book, person)
    .
    .

  @book = book
  book.rentals << self

  @person = person
  person.rentals << self
end
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @otmaneechchafyky 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ Added associations between classes 📌 
✔️ No linter errors 📌 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ghost — APPROVED** [review]
Hi @otmaneechchafyky , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Good work making the changes requested 👍 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**